core/renderers/measurables/row.js

```diff
@@ -28,207 +28,218 @@ const {Types} = goog.require('Blockly.blockRendering.Types');
-   * Does not include child blocks unless they are inline.
-   * @package
-   * @type {number}
-   */
-  this.width = 0;
-
-  /**
-   * The minimum height of the row.
-   * @package
-   * @type {number}
-   */
-  this.minHeight = 0;
-
-  /**
-   * The minimum width of the row, from the left edge of the block to the right.
-   * Does not include child blocks unless they are inline.
-   * @package
-   * @type {number}
-   */
-  this.minWidth = 0;
-
-  /**
-   * The width of the row, from the left edge of the block to the edge of the
-   * block or any connected child blocks.
-   * @package
-   * @type {number}
-   */
-  this.widthWithConnectedBlocks = 0;
-
-  /**
-   * The Y position of the row relative to the origin of the block's svg group.
-   * @package
-   * @type {number}
-   */
-  this.yPos = 0;
-
-  /**
-   * The X position of the row relative to the origin of the block's svg group.
-   * @package
-   * @type {number}
-   */
-  this.xPos = 0;
+   * @alias Blockly.blockRendering.Row
+   */
+  constructor(constants) {
+    /**
+     * The type of this rendering object.
+     * @package
+     * @type {number}
+     */
+    this.type = Types.ROW;
+
+    /**
+     * An array of elements contained in this row.
+     * @package
+     * @type {!Array<!Measurable>}
+     */
+    this.elements = [];
+
+    /**
+     * The height of the row.
+     * @package
+     * @type {number}
+     */
+    this.height = 0;
+
+    /**
+     * The width of the row, from the left edge of the block to the right.
+     * Does not include child blocks unless they are inline.
+     * @package
+     * @type {number}
+     */
+    this.width = 0;
+
+    /**
+     * The minimum height of the row.
+     * @package
+     * @type {number}
+     */
+    this.minHeight = 0;
+
+    /**
+     * The minimum width of the row, from the left edge of the block to the
+     * right. Does not include child blocks unless they are inline.
+     * @package
+     * @type {number}
+     */
+    this.minWidth = 0;
+
+    /**
+     * The width of the row, from the left edge of the block to the edge of the
+     * block or any connected child blocks.
+     * @package
+     * @type {number}
+     */
+    this.widthWithConnectedBlocks = 0;
+
+    /**
+     * The Y position of the row relative to the origin of the block's svg
+     * group.
+     * @package
+     * @type {number}
+     */
+    this.yPos = 0;
+
+    /**
+     * The X position of the row relative to the origin of the block's svg
+     * group.
+     * @package
+     * @type {number}
+     */
+    this.xPos = 0;
+
+    /**
+     * Whether the row has any external inputs.
+     * @package
+     * @type {boolean}
+     */
+    this.hasExternalInput = false;
+
+    /**
+     * Whether the row has any statement inputs.
+     * @package
+     * @type {boolean}
+     */
+    this.hasStatement = false;
+
+    /**
+     * TODO: Do not know what this does! FIX BEFORE MERGE.
+     * @type {number}
+     */
+    this.statementEdge = 0;
+
```

Comment:
@rachel-fenichel

Reply:
![image](https://user-images.githubusercontent.com/13686399/155597630-51dd3257-2b51-4e56-84a6-75fa6c260ff4.png)

I think it's what keeps the left sides of statement inputs aligned. In this case, adding an "else" adds padding to the right of each "do".

![image](https://user-images.githubusercontent.com/13686399/155597932-562b4137-c399-4d25-a2c9-ee0123f2c473.png)

Otherwise it's controlling where the right side of the bar above statement inputs goes, to make sure it's aligned for all statement inputs.
![image](https://user-images.githubusercontent.com/13686399/155598266-4008f10f-3fa4-446e-b843-b79822126e9e.png)

Reply:
Based on my reading it was the former :D Thank you Rachel!